truncate_client/src/lil_bits/battle.rs

```diff
@@ -17,69 +22,161 @@ impl<'a> BattleUI<'a> {
     }
 }
 
-fn render_word(battle_word: &BattleWord, ctx: &GameCtx, ui: &mut egui::Ui) {
-    let galley = ui.painter().layout_no_wrap(
-        battle_word.word.clone(),
+fn get_galleys<'a>(
+    battle_words: &'a Vec<BattleWord>,
+    ctx: &GameCtx,
+    ui: &mut egui::Ui,
+) -> Vec<Arc<Galley>> {
+    let dot = ui.painter().layout_no_wrap(
+        "•".into(),
         FontId::new(
-            ctx.theme.letter_size,
+            ctx.theme.letter_size * 0.75,
             egui::FontFamily::Name("Truncate-Heavy".into()),
         ),
-        match battle_word.valid {
-            Some(true) => ctx.theme.addition,
-            Some(false) => ctx.theme.defeated,
-            None => ctx.theme.outlines,
-        },
+        ctx.theme.outlines.darken(),
     );
-    let (rect, resp) = ui.allocate_at_least(galley.size(), Sense::hover());
-    ui.painter().galley(rect.min, galley);
 
-    resp.on_hover_ui(|ui| match (battle_word.valid, &battle_word.meanings) {
-        (None, _) => {
-            ui.label("Word did not need to be checked as the attack was invalid");
-        }
-        (Some(true), None) => {
-            ui.label("Valid word with no definition found");
-        }
-        (Some(true), Some(meanings)) => {
-            for meaning in meanings {
-                ui.label(format!("{}:", meaning.pos));
-                for def in &meaning.defs {
-                    ui.label(format!("  • {def}"));
-                }
+    let mut words: Vec<_> = battle_words
+        .iter()
+        .flat_map(|w| {
+            [
+                ui.painter().layout_no_wrap(
+                    w.word.clone(),
+                    FontId::new(
+                        ctx.theme.letter_size * 0.75,
```

Comment:
What's the `0.75` magic number about? Is it just that battle words are always gonna be a bit smaller than our theme size? Or is this a hacky thing to do with the font?

Reply:
> Is it just that battle words are always gonna be a bit smaller than our theme size?

Yeah that. Arguably we could have a bunch of font sizes in our theme that are more explicit, but maybe we go back through and backport the font sizes once everything is settled